struct.js

```diff
@@ -142,6 +144,32 @@ ThriftStruct.prototype.link = function link(model) {
         var field = this.fields[index];
         field.linkValue(model);
 
+        // Evidently with Apache Thrift, arguments are always optional,
+        // regardless of how they are marked.
+        // They are optional in Go by virtue of defaulting to the zero value
+        // for their type, and it is not possible to distinguish a missing
+        // field from the zero value.
+        if (this.isArgument) {
+            if (this.thrift.allowOptionalArguments) {
+                // Once this flag is enabled, all ThriftRW language
+                // implementations agree that fields are optional unless marked
+                // required.  If they are marked required, that contract is
+                // respected for both inbound and outbound messages.
+                if (!field.required && !field.optional && field.defaultValue === undefined) {
```

Comment:
Default values are inferred on the server side, when reading a struct with a missing field.

If a client uses the provided constructor for the struct, it will populate the default value as well, so in that sense the client also infers the default value.

However, clients are not obliged to use the constructor for the struct. In that case, if the value is missing, ThriftRW does not write it to the wire and instead depends on the server to infer the default.

Our libraries would be within their rights to detect that a given value matches the default and leave it out when writing. I have not done this and expect that "is it worth it" to depend on whether it is a primitive value or a compound object tree.